conngen/conngenmodule.cpp

```diff
@@ -281,34 +261,42 @@ ConnectionGeneratorModule::CGSelectImplementation_s_sFunction::execute(
    purposes. Usually, connections are created from a
    ConnectionGenerator using CGConnect.
 
+   Availability: Only if compiled with libneurosim support
    Author: Mikael Djurfeldt
    FirstVersion: March 2011
-   SeeAlso: CGParse, CGParseFile, CGConnect, CGSelectImplementation, cgstart,
-   cgnext
+   SeeAlso: CGParse, CGParseFile, CGConnect, CGSelectImplementation
 */
 void
-ConnectionGeneratorModule::CGSetMask_cg_iV_iVFunction::execute(
+ConnectionGeneratorModule::CGSetMask_cg_g_gFunction::execute(
   SLIInterpreter* i ) const
 {
   i->assert_stack_load( 3 );
 
   ConnectionGeneratorDatum cg =
     getValue< ConnectionGeneratorDatum >( i->OStack.pick( 2 ) );
-  IntVectorDatum sources = getValue< IntVectorDatum >( i->OStack.pick( 1 ) );
-  IntVectorDatum targets = getValue< IntVectorDatum >( i->OStack.pick( 0 ) );
+  GIDCollectionDatum sources =
+    getValue< GIDCollectionDatum >( i->OStack.pick( 1 ) );
+  GIDCollectionDatum targets =
+    getValue< GIDCollectionDatum >( i->OStack.pick( 0 ) );
+
+  RangeSet source_ranges;
+  cg_get_ranges( source_ranges, sources );
+
+  RangeSet target_ranges;
+  cg_get_ranges( target_ranges, targets );
```

Comment:
We never use `source_ranges` and `target_ranges`, except as argument to `cg_set_masks()`.  Wouldn't it be easier to just pass `sources` and `targets` to `cg_set_masks()`?

Reply:
Good catch. Fixed in fe4408921.